content/lectures/lec00.md

```diff
@@ -8,3 +8,73 @@ draft: false
 ---
 
 Hello world! This is the first lecture.
+
+## Demos
+
+### Poetry
+
+One reacurring design pattern we see in DevOps is package managers. This is a tool that helps manage your program's dependencies. In other words, the package manager is in charge of keeping track of what packages your project needs to run correctly, and then downloading those packages in a way that makes it easy for your program to use this auxillary code.
+
+We'll look at a few different package managers over the course of the semester. Node has one called NPM (Node Package Manager), Java has a package manager called Maven, and Python has a few offerings. Note that these package managers are all a little different, because they work with different languages that all have different nuances, this is why we can't reuse package managers across languages.
+
+The Python package manager we'll be using is called Poetry. Essentially, Poetry allows you to download certain Python libraries, then it creates a virtual python environment on your machine to run your code with the given libraries. So, why the virtual machine? The answer is that Python varies a lot from version to version (especially Python 2 compared to Python 3). The virtual environment ensures that you are running the same version of Python as your team of developers. It also allows us to ensure that the version of Python you run in development is the same as the version that runs in production.
+
+Now, let's get into how to actually use Poetry. First, make sure that you have Poetry installed on your machine, instructions for installation can be found [here](https://python-poetry.org/docs/).
+
+Once you have Poetry installed, let's create a new project:
+
+```
+# Create a new folder called poetry_demo
+$ mkdir poetry_demo
+# Enter the new folder
+$ cd poetry_demo
+$ poetry init
+```
+
+Now Poetry will give you lots of options for how to initialize your project, just hit enter for all of them (Poetry will go to the default setup which is fine for our purposes). Once you've finished, you'll see that there is a new file `pyproject.toml` in the directory, this is the file that stores the information we just initialized.
+
+Next, let's add a dependency:
+
+```
+$ poetry add redis
```

Comment:
you mean `poetry add numpy`?

Reply:
lol I switched packages halfway thank you